app/ui-react/syndesis/src/modules/integrations/pages/detail/DetailsPage.tsx

```diff
@@ -279,22 +372,39 @@ export class DetailsPage extends React.Component<
                                       )}
                                     />
                                     <IntegrationDetailHistoryListView
-                                      actions={draftActions}
+                                      actions={draftBtnActions}
                                       hasHistory={deployments.length > 0}
                                       isDraft={data.isDraft}
                                       i18nTextDraft={t('shared:Draft')}
                                       i18nTextHistory={t(
                                         'integrations:detail:History'
                                       )}
-                                      children={
-                                        <IntegrationDetailHistory
-                                          actions={actions}
-                                          deployments={deployments}
-                                          integrationId={data.id!}
-                                          updatedAt={data.updatedAt!}
-                                          version={data.version!}
-                                        />
-                                      }
+                                      children={deployments.map(
+                                        (deployment, idx) => {
+                                          return (
+                                            <IntegrationDetailHistoryListViewItem
+                                              key={idx}
+                                              actions={
+                                                <IntegrationDetailHistoryListViewItemActions
+                                                  actions={actions}
```

Comment:
Just so you're aware there's only a small subset of actions listed in the deployment list here, not the whole list:

![image](https://user-images.githubusercontent.com/351660/57041913-61b59c00-6c31-11e9-8408-3f99f9480b9d.png)

where "Replace Draft" I believe pulls the `flows` out of the deployment and replaces the one in the integration via a `PATCH` request.  It's done [like this](https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/integration/integration-actions-provider.service.ts#L287-L301) in the angular app, now I notice that it's actually buggy in there as it's not "flow" aware :-).  But the moar you know.  Then `publish` is the same as 'start' in the regular menu, but you use some details from the deployment instead.  The angular app does it in a hacky fashion [like this](https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/integration/integration_detail/integration-detail.component.ts#L164-L172)

Reply:
I see. I wonder then if it warrants adding two functions in `integrationFunctions.tsx`--one to replace draft and another for publish. I did automatically assume "publish" was just another word for "start". It changes things as bit as it means the dropdown menu is not 100% accurate. I think it'd be worth to create a separate issue for this though, as it's not directly related to the breadcrumbs. Let me know what you think.

Reply:
I'd say yeah, probably one for `replaceDraft` for sure.  The angular code re-used the same function for starting an integration, but yeah, I think it'd be better to have a separate `startDeployment` function.  [This](https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/core/providers/integration-support-provider.service.ts#L41-L57) is how it was done but I think a separate specific function would be cleaner.